algorithms/linfa-reduction/src/random_projection/common.rs

```diff
@@ -0,0 +1,11 @@
+/// Compute a safe dimension for a projection with precision `eps`,
+/// using the Johnson-Lindestrauss Lemma.
+///
+/// References:
+/// - [D. Achlioptas, JCSS](https://www.sciencedirect.com/science/article/pii/S0022000003000254)
+/// - [Li et al., SIGKDD'06](https://hastie.su.domains/Papers/Ping/KDD06_rp.pdf)
+pub(crate) fn johnson_lindenstrauss_min_dim(n_samples: usize, eps: f64) -> usize {
```

Comment:
You could add a test against values from [scikit-learn](https://scikit-learn.org/stable/modules/random_projection.html#the-johnson-lindenstrauss-lemma)

Reply:
Thank you for the suggestion, I added tests and found a bug!